hyphae/src/indexer.rs

```diff
@@ -0,0 +1,301 @@
+        let mut links = Vec::new();
+        let mut files: Vec<File> = self.files.values().map(|f| (*f).clone()).collect();
+        files.sort();
+        let mut taken: HashSet<String> = HashSet::new();
+        taken.insert("archive".to_string());
+        taken.insert("index.html".to_string());
+        let mut i = 0;
+        for f in files {
+            if html.len() > 1_000_000 {
+                warn!("index.html limited to fit in a single UnixFS node.");
+                break;
+            }
+            let mut real = f.name.clone();
+            while !taken.insert(real.clone()) {
+                real = i.to_string() + real.as_str();
+                i += 1;
+            }
+            let cid = Cid::try_from(f.cid.as_str());
+            if cid.is_err() {
+                continue;
+            }
+            let cid = cid?;
+            info!("Link: {}={}={}", &f.name, &real, &f.cid);
+            links.push(dag_pb::PbLink {
+                hash: Some(cid.to_bytes()),
+                name: Some(real.clone()),
+                tsize: None,
+            });
+            let when_str = f.when.to_rfc3339();
+            html.push_str(DATA_LINE_START);
+            html.push_str(&when_str);
+            html.push(' ');
+            html.push_str(&cid.to_string());
+            html.push(' ');
+            html.push_str(&f.name);
+            html.push_str(" -->\n");
+
+            html.push_str("<tr><td><a href='");
+            html.push_str(&real);
+            html.push_str("'>");
+            html.push_str(&f.name);
+            html.push_str("</a></td><td>");
+            html.push_str(&when_str);
+            html.push_str("</td></tr>\n");
+        }
+        html.push_str("</table></body></html>");
+        let html_bytes = Bytes::from(html);
+        let byte_count = html_bytes.len() as u64;
+        let node = UnixfsNode::Raw(html_bytes);
+        let index_block = node.encode()?;
+        links.push(dag_pb::PbLink {
+            hash: Some(index_block.cid().to_bytes()),
+            name: Some("index.html".to_string()),
+            tsize: Some(byte_count),
+        });
+        let inner = unixfs_pb::Data {
+            r#type: unixfs::DataType::Directory as i32,
+            ..Default::default()
+        };
+        let outer = ipfs_unixfs::builder::encode_unixfs_pb(&inner, links)?;
+        let node = UnixfsNode::Directory(unixfs::Node { outer, inner });
+        Ok([index_block, node.encode()?])
+    }
+}
+
+impl Ord for File {
+    fn cmp(&self, other: &Self) -> Ordering {
+        for o in &[self.when.cmp(&other.when).reverse(), self.name.cmp(&other.name)] {
+            if !o.is_eq() {
+                return *o;
+            }
+        }
+        self.cid.cmp(&other.cid)
+    }
+}
+
+impl PartialOrd<File> for File {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+        /*
```

Comment:
? could you leave a description of what this is for if the intent is to leave the comment?

Reply:
Removed comment.